testutils/check-calling-form-in-baseline.go

```diff
@@ -0,0 +1,210 @@
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// +build ignore
+
+package main
+
+import (
+	"bufio"
+	"flag"
+	"fmt"
+	"io"
+	"io/ioutil"
+	"log"
+	"os"
+	"path/filepath"
+	"regexp"
+	"sort"
+	"strings"
+)
+
+const usage = `Reads calling forms from gapic_yaml_file and make sure each is used by
+some *.baseline file in dir.
+The line specifying the calling form is in format
+
+  callingFormCheck: <id> <language>: <callingForm1> <callingForm2> ...
+
+The string "callingFormCheck" must start the line, preceded only by a possilby empty
+run of whitespaces.
+`
+
+func main() {
+	yamlFname := flag.String("yaml", "", "gapic yaml file")
+	flag.Usage = func() {
+		binName := os.Args[0]
+		out := os.Stderr // Should be flag.CommandLine.Output(), but Travis is too old
+		fmt.Fprintf(out, "Usage of %s:\n", binName)
+		fmt.Fprintf(out, "%s -yaml <gapic_yaml_file> [dir]\n", binName)
+		flag.PrintDefaults()
+		fmt.Fprint(out, usage)
+	}
+	flag.Parse()
+
+	if *yamlFname == "" {
+		flag.Usage()
+		os.Exit(1)
+	}
+	yamlf, err := os.Open(*yamlFname)
+	if err != nil {
+		log.Fatal(err)
+	}
+	defer yamlf.Close()
+
+	checks, err := readChecks(yamlf)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	walkFn := func(path string, info os.FileInfo, err error) error {
+		if err != nil {
+			return err
+		}
+		if len(checks) == 0 {
+			return filepath.SkipDir
+		}
+		if !info.Mode().IsRegular() {
+			return nil
+		}
+		if filepath.Ext(path) != ".baseline" {
+			return nil
+		}
+		return delFoundForms(path, checks)
+	}
+	dir := "."
+	if flag.NArg() > 0 {
+		dir = flag.Arg(0)
+	}
+	if err := filepath.Walk(dir, walkFn); err != nil {
+		log.Fatal(err)
+	}
+
+	if len(checks) > 0 {
+		var errs []checkConfig
+		for c := range checks {
+			errs = append(errs, c)
+		}
+		sort.Sort(checkConfigSlice(errs)) // Not sort.Slice, Travis is too old.
+		fmt.Println("not found:", errs)
+		os.Exit(1)
+	}
+}
+
+type checkConfigSlice []checkConfig
+
+func (s checkConfigSlice) Len() int      { return len(s) }
+func (s checkConfigSlice) Swap(i, j int) { s[i], s[j] = s[j], s[i] }
+func (s checkConfigSlice) Less(i, j int) bool {
+	if s[i].lang != s[j].lang {
+		return s[i].lang < s[j].lang
+	}
+	if s[i].id != s[j].id {
+		return s[i].id < s[j].id
+	}
+	return s[i].form < s[j].form
+}
+
+var checkPrefix = "callingFormCheck:"
+
+func readChecks(r io.Reader) (map[checkConfig]bool, error) {
+	sc := bufio.NewScanner(r)
+	checks := map[checkConfig]bool{}
+	for sc.Scan() {
+		line := sc.Text()
+		line = strings.TrimSpace(line)
+		if !strings.HasPrefix(line, checkPrefix) {
+			continue
+		}
+		line = line[len(checkPrefix):]
+
+		p := strings.IndexByte(line, ':')
+		if p < 0 {
+			goto fail
+		}
+		idLang := strings.Fields(line[:p])
+		if len(idLang) != 2 {
```

Comment:
It doesn't. The way I thought about it: If there are multiple languages, just write multiple `callingFormCheck` lines since different langs might have different forms anyway. We can revisit this if you'd like?

Reply:
OK, that should work, and it's more readable than what I originally had in mind.